client-v2/src/shared/bundles/collectionsBundle.ts

```diff
@@ -49,10 +58,59 @@ const collectionSelectors = {
   }
 };
 
+const SET_HIDDEN = 'SET_HIDDEN' as 'SET_HIDDEN';
```

Comment:
whuh?

Reply:
Weird hey? @jonathonherbert says this is some magic to make a string token type.

Reply:
Ohhh - kinda like `const SET_HIDDEN: 'SET_HIDDEN' = 'SET_HIDDEN'` I gotcha.

Reply:
https://basarat.gitbooks.io/typescript/docs/types/literal-types.html

Reply:
@sihil @itsibitzi Haha, yes, it's odd. I learned last week can now do this as of TS 3.4 with 'const assertions':

```const SET_HIDDEN = 'SET_HIDDEN' as const```ts

which at least removes the repetition.